### PR TITLE
👨🏻‍💼 Add private key sending transactions

### PR DESCRIPTION
**Pull request by mj426382**
closes https://github.com/TrueFiEng/useDApp/issues/702

### CODE REVIEW

File: packages/core/src/model/TransactionOptions.ts [hunk 1]
```diff
@@ -6,4 +6,9 @@ import { Signer } from 'ethers'
 export interface TransactionOptions {
   signer?: Signer
   transactionName?: string
+  privateKey?: string
+  mnemonicPhrase?: string
+  encryptedJson?: string
+  chainId?: number // privateKey, mnemonicPhrase and encryptedJso
```
Comment (rzadp) [inline packages/core/src/model/TransactionOptions.ts:12]: I don't understand the comment?

File: packages/core/src/model/TransactionOptions.ts [hunk 2]
```diff
@@ -6,4 +6,9 @@ import { Signer } from 'ethers'
...
+  password?: string // for encryptedJson purposes
```
Comment (rzadp) [inline packages/core/src/model/TransactionOptions.ts:13]: Hmm, how about something like:

```ts
encryptedJson?: { json: string, password: string }
```

File: packages/docs/docs/02-Guides/03-Transactions/08-Sendign Without Wallet.mdx [hunk 3]
```diff
@@ -0,0 +1,49 @@
+# Sending without Wallet
```
Comment (rzadp) [inline packages/docs/docs/02-Guides/03-Transactions/08-Sendign Without Wallet.mdx:1]: Typo in file name.

File: packages/core/src/hooks/useSendTransaction.test.ts [hunk 4]
```diff
@@ -51,22 +64,81 @@ describe('useSendTransaction', () => {
     expect(result.current.state.errorMessage).to.eq('invalid address')
   })
 
-  it('Returns receipt after correct transaction', async () => {
```
Comment (rzadp) [inline packages/core/src/hooks/useSendTransaction.test.ts:54]: Shouldn't we keep this test as well?

File: packages/core/src/model/TransactionOptions.ts (hunk 1 above)
Comment (mj426382) [reply packages/core/src/model/TransactionOptions.ts]: Solved :)

File: packages/core/src/model/TransactionOptions.ts [hunk 5]
```diff
@@ -6,4 +6,9 @@ import { Signer } from 'ethers'
 export interface TransactionOptions {
   signer?: Signer
   transactionName?: string
+  privateKey?: string
```
Comment (yivlad) [inline packages/core/src/model/TransactionOptions.ts:9]: Does signer and privateKey fields make sense being placed in the same object? If not maybe something like this:
```ts
export interface TransactionOptionBase {
  transactionName?: string
}
export interface TransactionOptionWithSigner {
  signer?: Signer
}
export interface TransactionOptionsWithPrivateKey {
  privateKey?: string,
  chainId?: number
}
...
export type TransactionOptions = TransactionOptionBase & (TransactionOptionWithSigner | TransactionOptionsWithPrivateKey | ...)
```

File: packages/core/src/helpers/getSignerFromOptions.ts [hunk 6]
```diff
@@ -0,0 +1,26 @@
+import { ethers, providers } from 'ethers'
```
Comment (mj426382) [inline packages/core/src/helpers/getSignerFromOptions.ts:1]: Add some test for this

**Review by yivlad — APPROVED** [review]

**Pull request merged**